robot_calibration/src/generate_and_publish_joint_states.cpp

```diff
@@ -1,28 +1,43 @@
 #include "ros/ros.h"
 #include "sensor_msgs/JointState.h"
+#include <stdlib.h>
+#include <time.h>
 
 int main(int argc, char **argv)
 {
   ros::init(argc, argv, "simple_joint_state_pub");
 
-  ros::NodeHandle nodeHandler;
+  ros::NodeHandle node;
 
-  ros::Publisher joint_states_pub = nodeHandler.advertise<sensor_msgs::JointState>("joint_states", 20);
+  ros::Publisher joint_states_pub = node.advertise<sensor_msgs::JointState>("joint_states", 20);
 
-  ros::Rate loop_rate(10);
-
-  while(ros::ok())
+  int num_messages;
+  node.param("num_messages", num_messages, 50);
+ 
+  double message_interval;
+  node.param("message_interval", message_interval, 0.1);
```

Comment:
This is a pretty small interval for updating the robot joint angles, and would be hard to observe in RViz. Please use a more reasonable default e.g. 3 seconds.

Also, as stated above, this should not be message interval, but duration in which to update the robot. 

Format the yaml something like the following if that is clearer (instead of message_interval and num_messages):
```
robot_joint_angles:
{
  num_new_positions: 50,
  update_interval: 5
}
```
When accessing the param, you would use "robot_joint_angles/update_interval". Feel free to give these better names.